app/controllers/restaurants_controller.rb

```diff
@@ -2,7 +2,15 @@ class RestaurantsController < ApplicationController
   before_action :set_restaurant, only: [:show, :update, :destroy]
 
   def index
-    render json: Restaurant.all
+    if request.query_parameters.present?
```

Comment:
You have the params available at `params`, a built-in method in Rails controller.
You should also ensure that only expected params are there, for security reasons (so no one passes funky params)

e.g. `permitted_params = params.permit(:my_first_query_param, :my_second_query_param)`

I suggest reading about the `permit`, `reject` and `require` methods.